main.go

```diff
@@ -97,14 +103,24 @@ func main() {
 		BurstSize: 100,
 	})
 
+	diagnosticsOpts := flags.GetDiagnosticsOptions(diagnosticsOptions)
```

Comment:
> lets add some logic to handle the case properly when user submits this flag vs the `diagnostics-address`

Similarly, this logic is being handled via [GetDiagnosticsOptions](https://github.com/kubernetes-sigs/cluster-api/blob/main/util/flags/diagnostics.go#L58-L94)
If both flags `--metrics-bind-addr` and `--diagnostics-address` are passed `--metrics-bind-addr` is given priority as of now.